### PR TITLE
docs: Improve id-denylist documentation

### PR DESCRIPTION
**Pull request by frankie303**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I added a small note to `id-denylist` documentation to clarify the usage of configuration array. It took me some time to realise the first element of the array specifies the severity of the rule even tough it also seems like one of the disallowed identifiers. So this small addition addresses the confusion. Also you can see [this issue](https://github.com/eslint/eslint/issues/13534) which mentioned the same confusion.

### CODE REVIEW

**Review by snitin315 — COMMENTED** [review]
A similar case was discussed in https://github.com/eslint/eslint/pull/11331, but it seems that we didn't open a follow-up issue to enhance the rule page template.

**Comment by frankie303** [comment]
> A similar case was discussed in #11331, but it seems that we didn't open a follow-up issue to enhance the rule page template.

I didn't see that issue before and I would definitely suggest to merge one of the PRs. It's clear that this rule causes some confusion for developers. Thanks!

File: docs/src/rules/id-denylist.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ For example, to restrict the use of common generic identifiers:
 }
 ```
 
+**Note:** First element of the array specifies the severity of the rule and it should be one of the following: 0 = "off", 1 = "warn", 2 = "error". Rest are the disallowed identifiers you want to add.
```
Comment (nzakas) [inline docs/src/rules/id-denylist.md:41]: Instead of restating the values for severity, can we just link to the section on configuring rules in the docs?

Comment (frankie303) [reply docs/src/rules/id-denylist.md]: Hi Nicholas, thanks for the feedback! I updated it. Please let me know if I can do further improvement in the sentence.

File: docs/src/rules/id-denylist.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ For example, to restrict the use of common generic identifiers:
 }
 ```
 
-**Note:** First element of the array specifies the severity of the rule and it should be one of the following: 0 = "off", 1 = "warn", 2 = "error". Rest are the disallowed identifiers you want to add.
+**Note:** First element of the array is for [rules configuration](https://eslint.org/docs/latest/user-guide/configuring/rules). Rest are the disallowed identifiers you want to add.
```
Comment (nzakas) [inline docs/src/rules/id-denylist.md:41]:
```suggestion
**Note:** The first element of the array is for the rule severity (see [configuring rules](/docs/latest/user-guide/configuring/rules). The other elements in the array are the identifiers that you want to disallow.
```

**Review by nzakas — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**